packages/datadog-instrumentations/src/graphql.js

```diff
@@ -186,7 +186,9 @@ function wrapExecute (execute) {
 }
 
 function wrapResolve (resolve) {
-  if (typeof resolve !== 'function' || patchedResolvers.has(resolve)) return resolve
+  if (typeof resolve !== 'function' || patchedResolvers.has(resolve) || !startResolveCh.hasSubscribers) {
```

Comment:
Since the configuration could be changed at runtime, the resolver should still be wrapped but the wrapper could return early when there are no subscribers. The presence of subscribers would then be checked on every call.